### PR TITLE
Updated to Jakarta JAXB and Java 17 (ref: #1075)

### PR DESCRIPTION
**Pull request by hilbertglm**
I changed the version in the POM to v8, since another issue indicated that the change would be a version change.  Other than that, I just used the Intellij IDEA Refactor/Migrate... to change the package names and updated the POMs accordingly.  No other functional code changes were made.  The unit tests passed.

### CODE REVIEW

**Comment by emckee2006** [comment]
Will this patch still work with the current Java level of the repo (11)?

**Comment by hilbertglm** [comment]
I didn't test it, because I assumed it would be a major release branch.  I ran it on Java 19 with a target level of 17.  I can do it when I get back to the office on Tuesday.  Note that I also changed the release in the POM.  If you want to reject the pull.  I can change the version back to 7.0.3-SNAPSHOT, and test against Java 11, and create a new pull request.  That might be the better way.

**Comment by hilbertglm** [comment]
I installed JDK on my local machine and ran `mvn clean test install` and that completed with no errors.  I changed the version back to 7.0.3-SNAPSHOT and the Java version from 17 to 11 in the POM.

File: pom.xml [hunk 1]
```diff
@@ -462,14 +462,14 @@
 				<version>1.039</version>
 			</dependency>
 			<dependency>
-				<groupId>javax.xml.bind</groupId>
-				<artifactId>jaxb-api</artifactId>
-				<version>2.3.1</version>
+				<groupId>jakarta.xml.bind</groupId>
+				<artifactId>jakarta.xml.bind-api</artifactId>
+				<version>4.0.0</version>
```
Comment (josemduarte) [inline pom.xml:467]: I think versions should be declared in this top pom.xml only (note this is the `dependencyManagement` section) and not in the modules. Then we have centralised version management.

File: biojava-structure/pom.xml [hunk 2]
```diff
@@ -67,15 +67,16 @@
...
 		</dependency>
 
-		<!-- JAXb explicit dependency is needed since Java 9. See https://stackoverflow.com/questions/43574426/how-to-resolve-java-lang-noclassdeffounderror-javax-xml-bind-jaxbexception-in-j
-		  Note versions are set in parent pom in dependency management section -->
 		<dependency>
-			<groupId>javax.xml.bind</groupId>
-			<artifactId>jaxb-api</artifactId>
+			<groupId>jakarta.xml.bind</groupId>
+			<artifactId>jakarta.xml.bind-api</artifactId>
+			<version>4.0.0</version>
```
Comment (josemduarte) [inline biojava-structure/pom.xml:73]: Please remove version here. It should be only in `dependencyManagement` in top pom.xml

File: biojava-structure/pom.xml [hunk 3]
```diff
@@ -67,15 +67,16 @@
...
 		</dependency>
 		<dependency>
 			<groupId>org.glassfish.jaxb</groupId>
 			<artifactId>jaxb-runtime</artifactId>
+			<version>4.0.3</version>
+			<scope>runtime</scope>
```
Comment (josemduarte) [inline biojava-structure/pom.xml:79]: Please remove version and scope here. It should be only in `dependencyManagement` in top pom.xml

File: biojava-protein-disorder/pom.xml [hunk 4]
```diff
@@ -85,12 +85,15 @@
 			<artifactId>log4j-core</artifactId>
 		</dependency>
 		<dependency>
-			<groupId>javax.xml.bind</groupId>
-			<artifactId>jaxb-api</artifactId>
+			<groupId>jakarta.xml.bind</groupId>
+			<artifactId>jakarta.xml.bind-api</artifactId>
+			<version>4.0.0</version>
```
Comment (josemduarte) [inline biojava-protein-disorder/pom.xml:90]: Please remove version here. It should be only in `dependencyManagement` in top pom.xml

File: biojava-protein-disorder/pom.xml [hunk 5]
```diff
@@ -85,12 +85,15 @@
...
 		</dependency>
 		<dependency>
 			<groupId>org.glassfish.jaxb</groupId>
 			<artifactId>jaxb-runtime</artifactId>
+			<version>4.0.3</version>
+			<scope>runtime</scope>
```
Comment (josemduarte) [inline biojava-protein-disorder/pom.xml:96]: Please remove version and scope here. It should be only in `dependencyManagement` in top pom.xml

File: biojava-modfinder/pom.xml [hunk 6]
```diff
@@ -54,12 +54,15 @@
 			<artifactId>log4j-core</artifactId>
 		</dependency>
 		<dependency>
-			<groupId>javax.xml.bind</groupId>
-			<artifactId>jaxb-api</artifactId>
+			<groupId>jakarta.xml.bind</groupId>
+			<artifactId>jakarta.xml.bind-api</artifactId>
+			<version>4.0.0</version>
```
Comment (josemduarte) [inline biojava-modfinder/pom.xml:59]: Please remove version here. It should be only in `dependencyManagement` in top pom.xml

File: biojava-modfinder/pom.xml [hunk 7]
```diff
@@ -54,12 +54,15 @@
...
 		</dependency>
 		<dependency>
 			<groupId>org.glassfish.jaxb</groupId>
 			<artifactId>jaxb-runtime</artifactId>
+			<version>4.0.3</version>
+			<scope>runtime</scope>
```
Comment (josemduarte) [inline biojava-modfinder/pom.xml:65]: Please remove version and scope here. It should be only in `dependencyManagement` in top pom.xml

File: biojava-core/pom.xml [hunk 8]
```diff
@@ -74,12 +74,15 @@
 			<artifactId>log4j-core</artifactId>
 		</dependency>
 		<dependency>
-			<groupId>javax.xml.bind</groupId>
-			<artifactId>jaxb-api</artifactId>
+			<groupId>jakarta.xml.bind</groupId>
+			<artifactId>jakarta.xml.bind-api</artifactId>
+			<version>4.0.0</version>
```
Comment (josemduarte) [inline biojava-core/pom.xml:79]: Please remove version here. It should be only in `dependencyManagement` in top pom.xml

File: biojava-core/pom.xml [hunk 9]
```diff
@@ -74,12 +74,15 @@
...
 		</dependency>
 		<dependency>
 			<groupId>org.glassfish.jaxb</groupId>
 			<artifactId>jaxb-runtime</artifactId>
+			<version>4.0.3</version>
+			<scope>runtime</scope>
```
Comment (josemduarte) [inline biojava-core/pom.xml:85]: Please remove version and scope here. It should be only in `dependencyManagement` in top pom.xml

File: biojava-aa-prop/pom.xml [hunk 10]
```diff
@@ -105,12 +105,15 @@
 		</dependency>
 
 		<dependency>
-			<groupId>javax.xml.bind</groupId>
-			<artifactId>jaxb-api</artifactId>
+			<groupId>jakarta.xml.bind</groupId>
+			<artifactId>jakarta.xml.bind-api</artifactId>
+			<version>4.0.0</version>
```
Comment (josemduarte) [inline biojava-aa-prop/pom.xml:110]: Please remove version here. It should be only in `dependencyManagement` in top pom.xml

File: biojava-aa-prop/pom.xml [hunk 11]
```diff
@@ -105,12 +105,15 @@
...
 		</dependency>
 		<dependency>
 			<groupId>org.glassfish.jaxb</groupId>
 			<artifactId>jaxb-runtime</artifactId>
+			<version>4.0.3</version>
+			<scope>runtime</scope>
```
Comment (josemduarte) [inline biojava-aa-prop/pom.xml:116]: Please remove version and scope here. It should be only in `dependencyManagement` in top pom.xml

**Review by josemduarte — CHANGES_REQUESTED** [review]
Thanks so much. Jaxb has been quite a headache as new versions of java rolled out. Hopefully this should give us some time to breathe. 

See comments about pom.

File: pom.xml (hunk 1 above)
Comment (hilbertglm) [reply pom.xml:467]: Removed explicit versions and scope from POM as required.

File: biojava-structure/pom.xml (hunk 2 above)
Comment (hilbertglm) [reply biojava-structure/pom.xml:73]: Removed explicit versions and scope from POM as required.

File: biojava-structure/pom.xml (hunk 3 above)
Comment (hilbertglm) [reply biojava-structure/pom.xml:79]: Removed explicit versions and scope from POM as required.

File: biojava-protein-disorder/pom.xml (hunk 4 above)
Comment (hilbertglm) [reply biojava-protein-disorder/pom.xml:90]: Removed explicit versions and scope from POM as required.

File: biojava-protein-disorder/pom.xml (hunk 5 above)
Comment (hilbertglm) [reply biojava-protein-disorder/pom.xml:96]: Removed explicit versions and scope from POM as required.

File: biojava-modfinder/pom.xml (hunk 7 above)
Comment (hilbertglm) [reply biojava-modfinder/pom.xml:65]: Removed explicit versions and scope from POM as required.

File: biojava-core/pom.xml (hunk 8 above)
Comment (hilbertglm) [reply biojava-core/pom.xml:79]: Removed explicit versions and scope from POM as required.

File: biojava-aa-prop/pom.xml (hunk 10 above)
Comment (hilbertglm) [reply biojava-aa-prop/pom.xml:110]: Removed explicit versions and scope from POM as required.

File: biojava-aa-prop/pom.xml (hunk 11 above)
Comment (hilbertglm) [reply biojava-aa-prop/pom.xml:116]: Removed explicit versions and scope from POM as required.

**Comment by hilbertglm** [comment]
Removed explicit scope and version, except in the dependency management section of the POM.  Re-compiled and ran unit tests with no error.

**Comment by hilbertglm** [comment]
It appears that openchart was removed from the JBoss Maven repository.  I looked and it isn't on Maven Central, either.  I Googled around and can't seem to find any mention of it.

**Comment by hilbertglm** [comment]
I took a look at my local repository, and openchart was there on May 26, 2023 when I first compiled biojava.  I looked inside the jar, and it was compiled November 28, 2008, so it appears to not be an active project.

**Review by josemduarte — APPROVED** [review]
Thanks for the changes! LGTM

Thanks also for checking the openchart dependency. I was fearing it would disappear one day from jboss repo. Since you've tested already, I think we can merge this for now and see how to get rid of the openchart dependency in another PR.

**Pull request merged**